frontend/src/lib/bare-metal-assets.ts

```diff
@@ -133,3 +105,101 @@ export async function attachBMAs(assets: JsonArray, hosts: JsonArray, clusterNam
         }
     })
 }
+
+export async function importBMAs() {
+    return new Promise((resolve, reject) => {
+        const input = document.createElement('input')
+        input.setAttribute('id', 'importBMAs')
+        input.type = 'file'
+        input.accept = '.csv, .txt'
+        input.style.visibility = 'hidden'
+        const body = document.querySelector('body')
+        body?.appendChild(input)
+        input.onchange = (e) => {
+            input.remove()
+            const file = e.target.files[0]
+            const reader = new FileReader()
+            reader.readAsText(file, 'UTF-8')
+            reader.onload = async (readerEvent) => {
+                const content = readerEvent.target.result
+
+                // parse csv
+                setTimeout(() => {
```

Comment:
How come a setTimeout is needed?

Reply:
replaced it with   
  await new Promise((r) => setTimeout(r, 5000));        
which I see is used in other places in console to pause the next functionality to let the ui thread finish